cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -0,0 +1,893 @@
+ where
+  -- The value is taken from a result of
+  -- script 'issue-genesis-utxo-expenditure.sh'.
+  networkMagic = CC.Common.NetworkTestnet 459045235
+
+-----------------------------------------------------------------------------------------
+-- Extract access to the Genesis funds.
+-----------------------------------------------------------------------------------------
+
+-- Extract the UTxO in the genesis block for the rich men
+extractGenesisFunds
+  :: CC.Genesis.Config
+  -> [Crypto.SigningKey]
+  -> Map Int (TxDetails, Crypto.SigningKey)
+extractGenesisFunds genesisConfig signingKeys =
+    Map.fromList
+  . mapMaybe (\(inp, out) -> mkEntry (inp, out) <$> isRichman out)
+  . fromCompactTxInTxOutList
+  . Map.toList
+  . CC.UTxO.unUTxO
+  . CC.UTxO.genesisUtxo
+  $ genesisConfig
+ where
+  mkEntry
+    :: TxDetails
+    -> (Int, Crypto.SigningKey)
+    -> (Int, (TxDetails, Crypto.SigningKey))
+  mkEntry txd (richman, key) = (richman, (txd, key))
+
+  isRichman :: CC.UTxO.TxOut -> Maybe (Int, Crypto.SigningKey)
+  isRichman out = listToMaybe $ filter (isValidKey . snd) richmen
+    where
+      isValidKey :: Crypto.SigningKey -> Bool
+      isValidKey key =
+        CC.Common.checkVerKeyAddress (Crypto.toVerification key) (CC.UTxO.txOutAddress out)
+
+  richmen :: [(Int, Crypto.SigningKey)]
+  richmen = zip [0..] signingKeys
+
+  fromCompactTxInTxOutList
+    :: [(CC.UTxO.CompactTxIn, CC.UTxO.CompactTxOut)]
+    -> [TxDetails]
+  fromCompactTxInTxOutList =
+    map (bimap CC.UTxO.fromCompactTxIn CC.UTxO.fromCompactTxOut)
+
+-----------------------------------------------------------------------------------------
+-- Work with initial funds.
+-----------------------------------------------------------------------------------------
+
+-- Prepare and submit our first transaction: send money from 'initialAddress' to 'sourceAddress'
+-- (latter corresponds to 'targetAddress' here) and "remember" it in 'availableFunds'.
+prepareInitialFunds
+  :: RunDemo (ByronBlockOrEBB ByronConfig)
+  => Tracer IO (TraceBenchTxSubmit (Byron.GenTxId (ByronBlockOrEBB ByronConfig)))
+  -> Tracer IO String
+  -> CardanoConfiguration
+  -> CC.Genesis.Config
+  -> NodeConfig ByronEBBExtNodeConfig
+  -> TopologyInfo
+  -> Map Int ((CC.UTxO.TxIn, CC.UTxO.TxOut), Crypto.SigningKey)
+  -> CC.Common.Address
+  -> CC.Common.Address
+  -> IO ()
+prepareInitialFunds _tracer
+                    llTracer
+                    cc
+                    genesisConfig
+                    pInfoConfig
+                    topologyInfo
+                    genesisUtxo
+                    genesisAddress
+                    targetAddress = do
+  let (mFunds, _, _initGenTx) = extractInitialFunds pInfoConfig
+                                                    (genesisUtxo Map.! 0) -- corresponds to 'genesisAddress'.
+                                                    targetAddress
+
+  let ((_, _), signingKey) = genesisUtxo Map.! 0
```

Comment:
I wonder if it'd be nicer to utilize `Map.!?`, pattern match on the result, and then explicitly call `error` if it's `Nothing`. This way we could also provide a more helpful error message detailing the module and function where the error occurred (i.e. `Cardano.CLI.Tx.Generation.prepareInitialFunds`). 

Also, it seems like we utilize `genesisUtxo Map.! 0` twice in this function so we could probably give it its own `let` binding.

@deepfire: think I'm nitpicking too much?

Reply:
> I wonder if it'd be nicer to utilize `Map.!?`, pattern match on the result, and then explicitly call `error` if it's `Nothing`.

It cannot be `Nothing` here, because of `extractGenesisFunds` function (which already called).